.github/workflows/ci.yml

```diff
@@ -218,47 +218,37 @@ jobs:
           secret_vault_address: ${{ secrets.VAULT_ADDR }}
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
 
-  java-unit-tests:
-    if: needs.detect-changes.outputs.java-code-changes == 'true'
-    name: "[UT] ${{ matrix.name }}"
+  # Sets up constants in unit tests. Using outputs instead of env vars so that these constants can be used in a matrix
+  setup-unit-tests:
     needs: [ detect-changes ]
```

Comment:
Could this job be conditional on the same `if: ..` that the later unit tests jobs depend on?